kolibri/plugins/coach/assets/src/modules/examCreation/actions.js

```diff
@@ -101,21 +101,24 @@ export function createExamAndRoute(store, classId) {
     learners_see_fixed_order: store.state.learnersSeeFixedOrder,
   };
 
-  store.commit('CORE_SET_PAGE_LOADING', true, { root: true });
-  createExam(store, exam).then(
-    () => {
-      router.push({ name: PageNames.EXAMS });
-      store.dispatch(
-        'createSnackbar',
-        {
-          text: snackbarTranslator.$tr('newExamCreated'),
-          autoDismiss: true,
-        },
-        { root: true }
-      );
-    },
-    error => store.dispatch('handleApiError', error, { root: true })
-  );
+  return new Promise((resolve, reject) => {
```

Comment:
Not sure that you need to wrap this `Promise` here, can just return `createExam` with no catch chained.